presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -620,6 +639,40 @@ public static long hash(long rawLo, long rawHi)
         return XxHash64.hash(rawLo) ^ XxHash64.hash(unpackUnsignedLong(rawHi));
     }
 
+    public static String toStringDestructive(Slice decimal)
+    {
+        if (isZero(decimal)) {
+            return "0";
+        }
+
+        String[] digitGroup = new String[MAX_PRECISION / MAX_POWER_OF_TEN_INT + 1];
```

Comment:
Given that it's for CAST, I don't think it matters a whole lot. I'd be more concerned if it was for primitive operations.